src/jit/lclvars.cpp

```diff
@@ -4026,10 +4026,8 @@ void Compiler::lvaMarkLocalVars()
 
             lvaTable[info.compLvFrameListRoot].lvType = TYP_I_IMPL;
 
-            /* Set the refCnt, it is used in the prolog and return block(s) */
-
-            lvaTable[info.compLvFrameListRoot].setLvRefCnt(2);
-            lvaTable[info.compLvFrameListRoot].setLvRefCntWtd(2 * BB_UNITY_WEIGHT);
+            // This local has implicit prolog and epilog references
+            lvaTable[info.compLvFrameListRoot].lvImplicitlyReferenced = 1;
```

Comment:
There is some logic in the (old/new) register allocation that tries to determine wheather it is worth spilling a caller saved register so that another LclVar can be enregistered.  If the weighted count is greater than 2*BB_UNITY_WEIGHT it is worth spilling.  if it is equal then its a wash but if it is less then we will stack allocate the LclVar.  

That is the purpose of setting the WtdRef count higher here.

Reply:
Hopefully down the road when we recompute ref counts before the RA runs, we'll see the actual number of references explicitly in the IR and not have to anticipate counts like this.

Reply:
The plan for those is to rely either on `lvImplicitlyReferenced` or some similar mechanism to figure out which locals may have implict refs and how much to count and weigh them.